dimod/binary/pybqm.py

```diff
@@ -84,10 +84,22 @@ def add_quadratic_from_dense(self, quadratic: ArrayLike):
         for u in range(num_variables):
             for v in range(num_variables):
                 if u == v:
-                    self.add_linear(u, quadratic[u, v])
+                    continue
```

Comment:
The modified code looks okay. But now the code under this elif condition caught my eye. You add a quadratic only if the quadratic exists ? What about if it does not exist. Secondly what if it exists and the value is positive 0, will the condition still be exited ?

Reply:
In this case we only add non-zero biases from the off diagonal, which makes sense since our representation is sparse.

Reply:
E.g.
```
In [5]: from dimod.binary.pybqm import pyBQM

In [6]: bqm = pyBQM('SPIN')

In [7]: bqm.add_quadratic_from_dense([[0, 0, 1], [0, 0, 0], [0, 0, 0]])

In [10]: list(bqm.iter_quadratic())
Out[10]: [(0, 2, 1)]
```
in this case only one interaction is added.

Reply:
Oh got it thanks.

Reply:
Just curious does python for an if condition consider if all the bits are zero or not , or it actually tries to find out if the value is zero or not. That is what if someone passses [-0 , -0 , -0] will it be added ?

Reply:
All of them evaluate False
```
In [11]: bool(0)
Out[11]: False

In [12]: bool(0.)
Out[12]: False

In [13]: bool(-0.)
Out[13]: False

In [14]: bool(-0)
Out[14]: False
```
so none would get picked up, which I think it fine